internal/provider/provider.go

```diff
@@ -221,3 +223,12 @@ func New(version string) func() provider.Provider {
 		}
 	}
 }
+
+func (p *NuoDbaasProvider) ConfigValidators(context.Context) []provider.ConfigValidator {
+	return []provider.ConfigValidator{
+		providervalidator.RequiredTogether(
+			path.MatchRoot("user"),
+			path.MatchRoot("password"),
```

Comment:
What if the user specifies the `user` attribute and the `NUODB_CP_PASSWORD` environment variable?

Reply:
This also wouldn't catch the case where only one of `NUODB_CP_USER`, `NUODB_CP_PASSWORD` is specified. I think we should just do explicit validation on this.